linalg/operator.cpp

```diff
@@ -365,6 +365,44 @@ void SecondOrderTimeDependentOperator::ImplicitSolve(const double dt0,
    mfem_error("SecondOrderTimeDependentOperator::ImplicitSolve() is not overridden!");
 }
 
+SumOperator::SumOperator(const Operator *A, const double alpha,
+                         const Operator *B, const double beta,
+                         bool ownA, bool ownB)
+   : Operator(A->Height(), A->Width()),
+     A(A), B(B), alpha(alpha), beta(beta), ownA(ownA), ownB(ownB),
+     a(A->Width()), b(B->Width())
```

Comment:
Shouldn't it be `A->Height()` here instead? Unless `MultTranspose` is being called. I think the semantics here need to be changed a bit.

Reply:
As in `ProductOperator`, `a` and `b` are in the range of `A` and `B`, respectively. I will remove one vector, and rename it to `z`, to avoid confusion with the previous idea.

Reply:
If those vectors are in the range, then their size should be `Height` not `Width` (but the transpose case also needs to be taken into account). It's OK to call `SetSize` in the `Mult` or MultTranspose`, once the vector has a certain capacity allocated, growing or shrinking its size (within that capacity) is cheap.